syntax = "proto3";

package anzen.v1;

enum DataType {
  DATA_TYPE_UNSPECIFIED = 0;
  DATA_TYPE_STRING = 1;
  DATA_TYPE_BINARY = 2;
  DATA_TYPE_FLOAT = 3;
  DATA_TYPE_INTEGER = 4;
}

message Data {
  optional string string_value = 1;
  optional bool binary_value = 2;
  optional double float_value = 3;
  optional sint64 int_value = 4;
}

enum ArmStatus {
  ARM_STATUS_UNSPECIFIED = 0;
  ARM_STATUS_ARMED = 1;
  ARM_STATUS_DISARMED = 2;
}

